app/users.go

```diff
@@ -97,6 +111,16 @@ func (s *UserService) GetUserByEmail(email string) (User, error) {
 	return user, nil
 }
 
+func (s *UserService) isEmailUsed(v *validation.Validator, email string) bool {
```

Comment:
This is the method I was referring to should be done in the `CreateUserRequest.validate`. The example is in the huge comment.